apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts

```diff
@@ -35,7 +35,11 @@ export class FederationHooks {
 	public static canAddFederatedUserToNonFederatedRoom(callback: (user: IUser | string, room: IRoom) => Promise<void>): void {
 		callbacks.add(
 			'federation.beforeAddUserAToRoom',
-			(params: { user: IUser | string }, room: IRoom): void => {
+			(params: { user: IUser | string; inviter?: IUser }, room: IRoom): void => {
+				if (!params || !params.user || !room || !params.inviter) {
```

Comment:
Should the callback really not run when inviter is undefined?

Reply:
yeah because for some reason this callback it was running when the user it is from keycloak

Reply:
Hey @MarcosSpessatto is that ok to keep this change? I see it fixes the issue, but I guess this just always causes the callback not to run (otherwise we'd throw the `error-cant-add-federated-users` error since `isAddingANewExternalUser` would return true).

Reply:
We should remove the `!params.inviter` as @pierre-lehnen-rc said since we are not using the inviter inside the method handling this callback. If you remove the `!params.inviter`, is this throwing an error anyway?

Reply:
Yeah, it was still throwing an error since we were sending the `internalUser` as a string. I changed some stuff so that an `IUser` is sent and this is now working without the `!params.inviter` check.